qark/scanner/scanner.py

```diff
@@ -68,14 +68,18 @@ def run(self):
     def _run_checks(self, plugins):
         """Run all the plugins (besides manifest) on every file."""
         current_file_subject = Subject()
-        for plugin in (observer_plugin for observer_plugin in plugins if isinstance(observer_plugin, PluginObserver)):
+        plugins = (observer_plugin for observer_plugin in plugins if isinstance(observer_plugin, PluginObserver))
+        for plugin in plugins:
             current_file_subject.register(plugin)
 
         for filepath in self.files:
             current_file_subject.notify(filepath)  # All the plugins will be running now
             # reset the plugin file data to None
             current_file_subject.reset()
 
+        for plugin in plugins:
```

Comment:
I believe at this point the generator is going to be empty so that this won't do anything.

Reply:
what do you mean generator mean?  In our test, this can help include in the issues found in other plugins.

Reply:
Sorry, I did not notice that the `plugins` was a generator. I changed it to a list and ran a test. It works now.